### ISSUE TITLE
Targeted Android Versions

### ISSUE THREAD

**Issue by ArchBlood** [humhub/app#189]
I believe we should be transparent about what versions of Android are or should be supported for the app, after running some personal tests for older versions of Android (i.e KitKat, Lollipop, Marshmallow) these versions do not seem to work correctly with the current version of the HumHub app, nor do I think we should as even Google no longer support these versions.

https://community.humhub.com/comment/perma?id=49194

**Comment by lisandi** [comment]
Very good idea to avoid that mess that happened with PHP years ago where
5.6 was a headache even 7.2 was out.

Honestly hosting companies and users both should be forced to regular
update their applications also for improvement and security reasons. And it
is even more important on mobile applications as otherwise this backward
compatibility is harming the future development. Android 14 is out and 12
runs still on most phones while 10 is often used in cars and those care
suppliers seem to be as reluctant to update as those hosting companies that
used php5.6 and did not get rid of it.

**Comment by luke-** [comment]
@PrimozRatej Do you have an overview here?

**Comment by PrimozRatej** [comment]
It's good to note that support for versions below API 24 (Android 8.0-9.0) is becoming rare. Developers might still accommodate older versions if there's a substantial user base on those versions. As of today everything lower than  9.0 Pie has less than 3% market share. (source: https://gs.statcounter.com/os-version-market-share/android)

For our case numbers look like this:
![image](https://github.com/humhub/app/assets/10835179/fcff8e3a-ce8b-4a49-852e-5e68c6c7dbee)

Newer Android versions generally provide better tools and improved functionality. Supporting older versions often means writing more code to achieve similar features available in newer versions. Moreover, ensuring that the app works across multiple Android versions requires more testing, adding to the development effort.

**Comment by ArchBlood** [comment]
> It's good to note that support for versions below API 26 (Android 8.0) is becoming rare. Developers might still accommodate older versions if there's a substantial user base on those versions. As of today everything lower than 9.0 Pie has less than 3% market share. (source: https://gs.statcounter.com/os-version-market-share/android)
> 
> For our case numbers look like this: <img alt="" width="250" height="250" src="https://private-user-images.githubusercontent.com/10835179/341182482-fcff8e3a-ce8b-4a49-852e-5e68c6c7dbee.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg4MjY1MjcsIm5iZiI6MTcxODgyNjIyNywicGF0aCI6Ii8xMDgzNTE3OS8zNDExODI0ODItZmNmZjhlM2EtY2U4Yi00YTQ5LTg1MmUtNWU2OGM2YzdkYmVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE5VDE5NDM0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIxOWRiZTRhYTg2ZDk0OTU3YzE1YTg0ODM0NDg4MzZjZTQwMzliZGNmNTg4YmZlNDQ0ZTRlNTg1YWNlZTQ4ZmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.MlKMhvtK9iAzqO8E3_UU6wPVG031nQ2EEH2d06q6_E0">
> 
> Newer Android versions generally provide better tools and improved functionality. Supporting older versions often means writing more code to achieve similar features available in newer versions. Moreover, ensuring that the app works across multiple Android versions requires more testing, adding to the development effort.
> 
> In theory, we could go as low as Android 5.0, but that would mean using older versions of libraries and most importantly older version of Android WebView. This could lead to issues because we are already using features on the web that are not yet implemented in the library. I would not recommend going lower than 8.0.

I agree, older versions of Android shouldn't be supported in this case, I'm not sure we should official support Android 8.x either, namely because we can't account for custom Android versions or versions that are maintained by other manufacturers. So in this case I would officially support Android 9 to 14, and unofficially support Android 8.x under a tag "Not recommended".

**Comment by PrimozRatej** [comment]
I created a simple test on Firebase Test Lab. It confirms that our app works with Android 8 (API level 26) and up.

Test results:
<img src="https://github.com/humhub/app/assets/10835179/f8e96cba-993f-4703-b542-1583fc6a6ef6" width="300" height="500" />

Maybe we could automate this process for every production release within the deployment process.
